plugins/modules/foreman_snapshot.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Manisha Singhal (ATIX AG)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_snapshot
+short_description: Manage Foreman Snapshots
+description:
+  - "Manage Foreman Snapshots for Host Entities"
+  - "This module can create, update, revert and delete snapshots"
+  - "This module requires the foreman_snapshot_management plugin set up in the server"
+  - "See: U(https://github.com/ATIX-AG/foreman_snapshot_management)"
+author:
+  - "Manisha Singhal (@Manisha15) ATIX AG"
+requirements:
+    - "apypie"
+options:
+  name:
+    description:
+      - Name of Snapshot
+    required: true
+  description:
+    description:
+      - Description of Snapshot
+    required: false
+  host:
+    description:
+      - Name of related Host
+    required: true
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: "Create a Snapshot"
+  foreman_snapshot:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_snapshot"
+    host: "host_name"
```

Comment:
could you use more descriptive names? maybe something like:

```yaml
name: "snapshot_before_upgrade"
host: "server.example.com"
```

Reply:
We don't really change the name of snapshot once it is created so isn't "snapshot_before_upgrade" snapshot name a bit misleading ?

Reply:
I guess, it is meant to be "snapshot_before_major_software_upgrade".
(Not the snapshot itself is going to be updated)

Reply:
Yeah, I didn't mean to rename the snapshot, just wanted to make the snapshot name for all examples closer to the real world where you usually would not name your snapshot "new_snapshot".

Reply:
Ok. I have modified the module with suggested changes. It is ready for review now.